packages/store/src/operators/wrap.ts

```diff
@@ -0,0 +1,58 @@
+import { NgZone } from '@angular/core';
+
+import {
+  MonoTypeOperatorFunction,
+  Observable,
+  Operator,
+  Subscriber,
+  Subscription
+} from 'rxjs';
+
+/**
+ * Returns operator based on the provided condition `outsideZone`, that will run
+ * `subscribe` inside or outside Angular's zone
+ */
+export function wrap<T>(
+  outsideZone: boolean | null,
+  zone: NgZone
+): MonoTypeOperatorFunction<T> {
+  return (source: Observable<T>) => source.lift(new WrapOperator(outsideZone, zone));
+}
+
+class WrapOperator<T> implements Operator<T, T> {
+  constructor(private outsideZone: boolean | null, private zone: NgZone) {}
+
+  public call(subscriber: Subscriber<T>, source: Observable<T>): Subscription {
+    return source.subscribe(new WrapSubscriber(subscriber, this.outsideZone, this.zone));
+  }
+}
+
+class WrapSubscriber<T> extends Subscriber<T> {
```

Comment:
What is the reason for the vastly different approach to what was used with the enterZone pipe?
This is a lot more code.

Reply:
@markwhitfeld I can re-write `wrap` operator in the same way as `enterZone` was implemented

Reply:
https://github.com/ngxs/store/issues/739
https://github.com/ngxs/store/issues/487

in that we can now control the management of the zone

This feature will allow you to disable extra detect changes in angular.

Reply:
revert by https://github.com/ngxs/store/pull/763